src/game/server/entities/character.cpp

```diff
@@ -978,10 +978,15 @@ void CCharacter::StopRecording()
 
 void CCharacter::Die(int Killer, int Weapon, bool SendKillMsg)
 {
+	int ModeSpecial = GameServer()->m_pController->OnCharacterDeath(this, (Killer < 0) ? nullptr : GameServer()->m_apPlayers[Killer], Weapon);
+	
+	// insta: provide a way to cancel a death, either to do something else or modify the death
+	if(ModeSpecial == -1)
+		return;
```

Comment:
As discussed in game I am not a big fan of this. I was actually planning to do the exact same change but I would like to first think about it more and do it my self when I find time to properly think about it and document it.

Also would like to at least get https://github.com/ddnet/ddnet/pull/10727 closed/merged upstream first before touching that method.

Reply:
Thats done now :partying_face: see 50c90d74487d3aada6110ab2bcb4c26ac6b29516 you have now full control over the death message in the controller.